lib/Conversion/TosaToEmitC/TosaToEmitC.cpp

```diff
@@ -768,6 +768,8 @@ void populateTosaToEmitcPatterns(MLIRContext *ctx,
                                                            "emitc::tosa::add");
   patterns.add<ArithmeticRightShiftOpConversion>(
       ctx, "emitc::tosa::arithmetic_right_shift");
+  patterns.add<CallOpBroadcastableConversion<tosa::EqualOp>>(
+      ctx, "emitc::tosa::equal");
```

Comment:
With the current patch
```mlir
%0 = "tosa.equal"(%arg0, %arg1) : (tensor<13x21x3xi32>, tensor<13x21x3xi32>) -> tensor<13x21x3xi1>
```
is translated into

```C++
  Tensor<bool, 13, 21, 3> v3 = emitc::tosa::equal(v1, v2);
```
Looking into the header-only implementation you're providing with this PR, the return type needs to be passed as a template argument. This is currently not the case. However, the `CallOpConversion` as well as the `CallOpBroadcastableConversion` can handle this. With
```suggestion
      ctx, "emitc::tosa::equal", /*explicitResultType=*/true);
```
the mlir snippet is translated to:
```
Tensor<bool, 13, 21, 3> v3 = emitc::tosa::equal<Tensor<bool, 13, 21, 3>>(v1, v2);
```